server/src/main/java/org/opensearch/gateway/ShardsBatchGatewayAllocator.java

```diff
@@ -251,6 +251,16 @@ else if (shardRouting.primary() == primary) {
             }
         });
 
+        allocation.routingNodes().forEach(
+            routingNode ->
+            routingNode.getInitializingShards().forEach(
+                shardRouting -> {
+                    if (currentBatchedShards.containsKey(shardRouting.shardId()) && shardRouting.primary() == primary) {
+                        batchedShardsToAssign.add(shardRouting.shardId());
+                    }
+                })
+        );
+
```

Comment:
Still requires updating the ShardRouting in batch. @shiv0408

Reply:
Done in latest commit 👍🏼

Reply:
This is update, please resolve it